src/lib/irap_export_binary.cpp

```diff
@@ -36,22 +44,33 @@ void write_values_binary(surf_span values, std::ostream& out) {
   auto rows = values.extent(0);
   auto cols = values.extent(1);
   auto len = values.size();
+  constexpr long line_size = (PER_LINE_BINARY + 2) * sizeof(float);
+  constexpr size_t buflen = line_size * 10;
+  char buf[buflen];
+  char* bufptr = buf;
   for (size_t j = 0; j < cols; j++) {
     for (size_t i = 0; i < rows; i++) {
       if (written_on_line == 0) {
         chunk_length = std::min(len - (j * cols + i), PER_LINE_BINARY);
-        write_32bit_binary_value(out, chunk_length * 4);
+        write_32bit_binary_value(bufptr, chunk_length * 4);
       }
 
       auto v = values(i, j);
-      write_32bit_binary_value(out, std::isnan(v) ? UNDEF_MAP_IRAP : v);
+      write_32bit_binary_value(bufptr, std::isnan(v) ? UNDEF_MAP_IRAP : v);
 
       if (++written_on_line == chunk_length) {
-        write_32bit_binary_value(out, chunk_length * 4);
+        write_32bit_binary_value(bufptr, chunk_length * 4);
         written_on_line = 0;
       }
+
+      if (auto dist = std::distance(buf, bufptr); buflen - dist < line_size) {
```

Comment:
our of curiosity what is std:distance? :) 
Otherwise look good!

Reply:
It gives number of "hops" between buf and bufptr. Since buf is an array of chars the hops will be 1 char.

Reply:
https://en.cppreference.com/w/cpp/iterator/distance

> Returns the number of hops from first to last.

Ie, the number of elements (in this case `char`s) between `buf` and `bufptr`.

I had a huddle with Jon where I requested some changes.